src/teleop/gui/src/components/ROSService.vue

```diff
@@ -0,0 +1,322 @@
+                <option value="" selected>Select a service</option>
+                <option v-for="option in service_options" v-bind:value="option">
+                {{ option }}
+                </option>
+                </select>
+
+                <textarea v-if="args != '' && args != '{}'" class="box" id="textarea" v-model="args"></textarea>
+
+                <p> {{schema}} </p>
+                <p v-if="error">JSON Syntax Error! Cannot send...</p>
+
+                <button class="box" id="send" type="button" v-on:click="sendArgs()">Send</button>
+
+                </div>
+
+                <p class="box responseCell">{{response}}</p>
+            </div>
+      </div>
+    </div>
+  </template>
+  
+  <script>
+  
+  import ROSLIB from "roslib"
+  
+  
+  export default {
+    name: 'ROSService',
+    mounted() {
+      this.populateServices();
+    },
+    data() {
+        return {
+            service_options: [],
+            selectedService: '',
+            selectedType: '',
+            args: '',
+            schema: '',
+            error: false,
+            response: ''
+        }
+    },
+    
+
+    methods: {
+        populateServices: function() {
+          var topicsClient = new ROSLIB.Service({
+              ros : this.$ros,
+              name : '/rosapi/services',
+              serviceType : 'rosapi/Services'
+          });
+
+          var servicesSorted = [];
+          var request = new ROSLIB.ServiceRequest();
+          topicsClient.callService(request, (result) => {
+              servicesSorted = result.services.sort();
+              for(var r of servicesSorted){
+                this.service_options.push(r);
+              }
+          });
+
+        },
+
+        switchService : function(){
+            this.args = '';
+            this.schema = '';
+            this.error = false;
+            var serviceTypeClient = new ROSLIB.Service({
+              ros : this.$ros,
+              name : '/rosapi/service_type',
+              serviceType : 'ServiceType'
+            });
+
+            var request = new ROSLIB.ServiceRequest({service: this.selectedService});
+            serviceTypeClient.callService(request, (result) => {
+                this.selectedType = result.type;
+                getArgs(result.type)
+            });
+
+            const getArgs = (type) => {
+                var serviceClient = new ROSLIB.Service({
+                ros : this.$ros,
+                name : '/rosapi/service_request_details',
+                serviceType : 'ServiceRequestDetails'
+                });
+
+                var request1 = new ROSLIB.ServiceRequest({type: type});
+                serviceClient.callService(request1, (result) => {
+                    const displayArgs = (arr) => {
+                        for(var i = 0; i < arr.fieldtypes.length; i++){
+                            if(!(['bool', 'int8', 'uint8', 'int16', 'uint16', 
```

Comment:
Make this array a const declared near the imports and then use it here, having a large hardcoded array like this is tuff